### PR TITLE
feat(widget-frame): add fullSize toggle function

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

https://user-images.githubusercontent.com/50662170/210308436-ffb3f637-d44f-4bce-adae-385f6e1bca1c.mov

Expand button is disabled if the default size option does not contain 'full' or only 'full'.

### CODE REVIEW

**Review by minseolee — APPROVED** [review]
LGTM! Thank you very much 🚀

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 1]
```diff
@@ -167,12 +169,25 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
         }
     };
 
+    const setFullSizeWidget = (widgetKey: string) => {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:172]: nitpick: please specify return type!

File: src/services/dashboards/widgets/use-widget-state.ts [hunk 2]
```diff
@@ -73,6 +73,7 @@ export function useWidgetState<Data = any>(
         granularity: computed<Granularity>(() => state.widgetConfig.options?.granularity),
         chartType: computed(() => state.options.chart_type ?? CHART_TYPE.LINE),
         disableFullSize: computed<boolean>(() => !state.widgetConfig.sizes.includes(WIDGET_SIZE.full)),
+        isOnlyFullSize: computed<boolean>(() => state.widgetConfig.sizes.length === 1 && state.widgetConfig.sizes[0] === WIDGET_SIZE.full),
```
Comment (yuda110) [inline src/services/dashboards/widgets/use-widget-state.ts:76]: suggestion: I think this state can be moved to `useWidgetFrameProps()`.

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts (hunk 1 above)
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:172]: suggestion: How about `toggleWidgetSize`?

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 3]
```diff
@@ -167,12 +169,25 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
...
+        state.dashboardWidgetInfoList = state.dashboardWidgetInfoList.map((info) => {
+            const widgetSizes = getWidgetConfig(info.widget_name)?.sizes;
+            if (info.widgetKey === widgetKey && widgetSizes) {
+                return {
+                    ...info,
+                    size: (info.size === WIDGET_SIZE.full) ? (widgetSizes[0] ?? WIDGET_SIZE.md) : WIDGET_SIZE.full,
+                };
+            }
+            return info;
+        });
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:182]: suggestion: How about to change only target widget? (Code is just sample..)

```suggestion
        const _dashboardWidgetInfoList = cloneDeep(state.dashboardWidgetInfoList);
        const _targetWidgetIndex = _dashboardWidgetInfoList.findIndex((d) => d.widgetKey === widgetKey);
        if (_targetWidgetIndex > -1) {
            const _targetWidget = _dashboardWidgetInfoList[_targetWidgetIndex];
            const widgetSizes = getWidgetConfig(_targetWidget.widget_name)?.sizes;
            _dashboardWidgetInfoList[_targetWidgetIndex] = {
                ..._targetWidget,
                size: (_targetWidget.size === WIDGET_SIZE.full) ? (widgetSizes[0] ?? WIDGET_SIZE.md) : WIDGET_SIZE.full,
            }
            state.dashboardWidgetInfoList = _dashboardWidgetInfoList;
        }
```

**Review by yuda110 — APPROVED** [review]
LGTM! I left some comments

**Pull request merged**